python/pyarrow/tests/parquet/test_encryption.py

```diff
@@ -530,3 +530,45 @@ def kms_factory(kms_connection_configuration):
             path, decryption_properties=file_decryption_properties)
         result_table = result.read(use_threads=True)
         assert data_table.equals(result_table)
+
+
+def test_encrypted_parquet_read_table(tempdir, data_table, basic_encryption_config):
+    """Write an encrypted parquet then read it back using read_table."""
+    path = tempdir / 'encrypted_table_read_table.parquet'
```

Comment:
This is copy-pasting almost all the same code from other test functions, can you please factor out the common steps?